logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/ProtoContentMapper.kt

```diff
@@ -275,6 +275,21 @@ class ProtoContentMapperImpl(
         }
     }
 
+    private fun packEdited(readableContent: MessageContent.TextEdited): GenericMessage.Content.Edited {
+        val mentions = readableContent.newMentions.map { messageMentionMapper.fromModelToProto(it) }
+        return GenericMessage.Content.Edited(
+            MessageEdit(
+                replacingMessageId = readableContent.editMessageId,
```

Comment:
Probably a dummy question, but shouldn't we keep the original message id?

Reply:
Every edit changes the message id, I think the dilemma would be in case when you edit the same message on two clients and one of them fails - which one to choose as a final one then, for now we agreed that if the message edit fails we should already show the updated content locally on that one client, for now it will work because we don't have the "retry" mechanism so we won't be able to send an update with outdated id, but I think it's something worth discussing - how other platforms do that, should we keep all previous ids of such message in that case or maybe the backend handles that for us. 🤔

Reply:
This is problematic already now, consider this:

1. AR client1 edits message M1 but it fails
2. AR client2 also edits message M1 and succeds
3. Upon processing the incoming edit on client1 it can't apply the edit but the message ID has already been updated with the failed edit.

How can we solve this?

### Idea 1
When sending an edit, don't update the message ID until message edit signal message has been accepted by the backend.

Reply:
Yep, I checked that by editing message at the same time on web and scala app and it created two different messages in the end so it's not handled on other platforms as well.

> When sending an edit, don't update the message ID until message edit signal message has been accepted by the backend.

@typfel but then it will behave like this:
- we have a message with text "A"
- on device 1 we update it to "B" and it fails
- on device 2 we update it to "C" and it succeeds
- we get internet back on device 1, it receives the update from device 2 and changes the text of this message to "B"
- we lose the edit that we did on device 1

Reply:
I think the question is: what should be more important in this scenario - the pending edit that user created earlier on device 1 or newer edit from device 2 already accepted by backend and it should update the previous failed update?

Reply:
Yes, we'll lose the edit but at least everything is consistent.

If we want improve this we would need to introduce something like an edit draft which is stored separately from the original message we want to edit. This is way we could cancel a failed edit and revert back to the original message. But this probably out scope for the release.

But if even if have an edit draft it's still important to only update original message ID after an edit has been accepted by the. BE otherwise we run into consistency issues when messages are edited multiple devices.

Reply:
> I think the question is: what should be more important in this scenario - the pending edit that user created earlier on device 1 or newer edit from device 2 already accepted by backend and it should update the previous failed update?

Yep, this might need input from product. 

My hunch is that it's fine to overwrite failed edits since it's the same person doing the edits, so he/she actively decides edit again on another device and would expect that to have the same state synchronised across all devices after that.

Reply:
Ok, you're right.
I think we could improve it for now by changing if the incoming edit timestamp is older or newer than our local pending one - if newer then we use the incoming one, if older then we update the message id only and keep the content of local pending one, but maybe that's an overkill for now. 😅